mozilla/foundation.mozilla.org
# Buyers Guide campaign page

# Description



Related PRs/issues: #9082


This PR introduces the following:
- A new buyers guide page model, `BuyersGuideCampaignPage`, which is very similar to the original CampaignPage model, with some uneccesary fields removed.
- A new relation model `BuyersGuideCamppaignPageModalRelation`, which allows users to link DonationModals to the page.
- Accompanying migrations for both of the models
- New Factories for both models
- A test file `test_campaign_page.py` which tests:
  - Model Factories
  - Basic Page Functionality
  - The BuyersGuideCampaignPage's `get_donation_modal_json` method
- New HTML template `buyersguide/campaign_page.html`
- Updated some petition and CTA related  JS/CSS files from being Foundation site specific, to now being share dbetween foundation and the buyersguide

# Steps to test 

1. Check out this branch and run `inv new-db` in the command line
2. 
3. Navigate to the buyers guide article index page in the CMS, and note that there are new `BuyersGuideCampaignPage`s at the bottom of the list.
4. Edit an existing page, or create a new one from scratch.
5. The CMS should work as expected, allowing you to select a CTA, donation modals if desired, and streamfields for the body.
6. Visit the page on the front end and confirm the page data is being displayed
7. Visit the Editorial Content Index page and confirm the Campaign pages are being rendered in the list (Refined styling to be taken care of in future ticket)
8. If everything works as expected, testing is complete! 

# Screenshots 

Campaign Page Template:
![Screenshot 2022-10-17 at 23-59-22 Privacy Not Included A Buyer’s Guide for Connected Products](https://user-images.githubusercontent.com/18314510/196358593-5ca6fe28-cd56-451b-a2af-cff198d46882.png)

Campaign Page in Editorial Content Index list:
![Screenshot 2022-10-17 at 23-59-31 Privacy Not Included A Buyer’s Guide for Connected Products](https://user-images.githubusercontent.com/18314510/196358864-5ddb1d78-a6e1-45ef-a871-a45d344e10c7.png)

# Checklist



**Tests**

**Changes in Models:**

**Documentation:**
